cmd/show/tui.go

```diff
@@ -0,0 +1,87 @@
+package show
+
+import (
+	"fmt"
+	"os"
+
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+// ██████╗ ██╗███████╗███████╗ █████╗
+// ██╔══██╗██║╚══███╔╝╚══███╔╝██╔══██╗
+// ██████╔╝██║  ███╔╝   ███╔╝ ███████║
+// ██╔═══╝ ██║ ███╔╝   ███╔╝  ██╔══██║
+// ██║     ██║███████╗███████╗██║  ██║
+// ╚═╝     ╚═╝╚══════╝╚══════╝╚═╝  ╚═╝
+
+const (
+	dashboardView = iota
+	contributorView
+)
+
+// sessionState serves as the variable to reference when looking for which model the user is in
+type (
+	sessionState int
+	SuccessMsg   struct{}
+)
+
+type MainModel struct {
+	state       sessionState
+	dashboard   tea.Model
+	contributor tea.Model
+	authorName  string
+}
+
+// View: the view of the TUI
+func (m MainModel) View() string {
+	switch m.state {
+	case contributorView:
+		return m.contributor.View()
+	default:
+		return m.dashboard.View()
+	}
+}
+
+// Init: initial IO before program start
+func (m MainModel) Init() tea.Cmd { return nil }
+
+// Update: Handle IO and Commands
+func (m MainModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var cmd tea.Cmd
+	var cmds []tea.Cmd
+	switch msg := msg.(type) {
+	case BackMsg:
+		m.state = dashboardView
+	case SelectMsg:
+		m.state = contributorView
+		m.authorName = msg.contributorName
+	}
+
+	switch m.state {
+	case dashboardView:
+		newDashboard, newCmd := m.dashboard.Update(msg)
+		m.dashboard = newDashboard
+		cmd = newCmd
+	case contributorView:
+		// TODO: process cmd error if contributor fails to fetch
+		m.contributor, _ = InitContributor(m.authorName)
+		_, newCmd := m.contributor.Update(msg)
+		cmd = newCmd
+	}
+	cmds = append(cmds, cmd)
+	return m, tea.Batch(cmds...)
+}
+
+func pizzaTUI(opts *Options) {
+
+	// TODO: handle error to cobra
+	dashboardModel, _ := InitDashboard(opts)
+
+	model := MainModel{dashboard: dashboardModel}
+
+	if _, err := tea.NewProgram(model, tea.WithAltScreen()).Run(); err != nil {
+		fmt.Println("Error running program:", err)
+		os.Exit(1)
```

Comment:
Ok, yeah this makes sense. Normally for bubbletea only programs that is the standard way, but here since we are interfacing with cobra it makes sense to bubble the error to it.